dashboard/src/components/AppView/AppView.tsx

```diff
@@ -118,7 +116,7 @@ class AppView extends React.Component<IAppViewProps, IAppViewState> {
     }
     if (nextProps.error) {
       // close any existing sockets
-      this.closeSockets();
+      // this.closeSockets();
```

Comment:
where do you plan to close sockets? I don't see a clear "easy" way, maybe this component is the best place to do it.

Reply:
I think each component that opens a watch in componentDidMount should close the socket in componentWillUnmount. If this a close action gets executed multiple times by different components, the reducer will just ignore if the socket is already closed/non-existent.

This obviously has the caveat you mentioned in our call that a socket might be closed by a component but still in use by another component. This is not a problem today, but one way to fix this would be to use a counter to track uses of a socket and only close it once it reaches 0. This is something that we can add in the future when necessary.

Reply:
:+1:  sounds good